### PR TITLE
Add api to calculate scores for a given location

### PR DESCRIPTION
**Pull request by lemonlatte**
In this PR, we add a `/score` API for query scores using addresses. This API need the [nominatim](https://wiki.openstreetmap.org/wiki/Nominatim) api infrastructure to be prepared.

Changes Highlight:

- Add a nominatim client in external package
- Create an interface `LocationSearcher` to query coordinates by addresses

### CODE REVIEW

File: external/nominatim/client.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		RawQuery: url.Values{
+			"q":      []string{query},
+			"format": []string{"json"},
+		}.Encode(),
+	}
+
+	reqString := fmt.Sprintf("%s/%s", n.endpoint, q.String())
+	log.WithField("prefix", "nominatim").WithField("req", reqString).Debug("request from nominatim")
+
+	resp, err := n.client.Get(reqString)
```
Comment (nodestory) [inline external/nominatim/client.go:51]: Also check `resp.StatusCode`.

File: api/score.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/bitmark-inc/autonomy-api/geo"
+	"github.com/bitmark-inc/autonomy-api/schema"
+	"github.com/bitmark-inc/autonomy-api/score"
+	"github.com/gin-gonic/gin"
```
Comment (nodestory) [inline api/score.go:9]: The import order is messed up.

Comment (lemonlatte) [reply api/score.go]: Fixed

File: external/nominatim/client.go (hunk 1 above)
Comment (lemonlatte) [reply external/nominatim/client.go:51]: Checked

**Pull request merged**